File: src/main/java/com/github/kindrat/cassandra/client/util/JsonUtil.java
package com.github.kindrat.cassandra.client.util;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.type.CollectionType;
import lombok.SneakyThrows;
import lombok.experimental.UtilityClass;

import java.util.List;

@UtilityClass
public class JsonUtil {
    private static final ObjectMapper mapper = new ObjectMapper();

    @SneakyThrows
    public static String toJson(Object pojo) {
        return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(pojo);
    }

    @SneakyThrows
    public static <T> List<T> fromJson(String json, Class<T> type) {
        CollectionType mapType = mapper.getTypeFactory().constructCollectionType(List.class, type);
        return mapper.readerFor(mapType).readValue(json);
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/util/JsonUtil.kt
package com.github.kindrat.cassandra.client.util

import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper

object JsonUtil {
    private val mapper = jacksonObjectMapper()

    @JvmStatic
    fun toJson(pojo: Any?): String {
        return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(pojo)
    }

    @JvmStatic
    fun <T> fromJson(json: String?, type: Class<T>?): List<T> {
        val mapType = mapper.typeFactory.constructCollectionType(MutableList::class.java, type)
        return mapper.readerFor(mapType).readValue(json)
    }
}